contracts/core/State.sol

```diff
@@ -13,6 +13,7 @@ import "../openzeppelin/Ownable.sol";
 import "../openzeppelin/SafeMath.sol";
 import "../interfaces/IWrbtcERC20.sol";
 import "../reentrancy/SharedReentrancyGuard.sol";
+import "../reentrancy/LoanIdGuard.sol";
```

Comment:
optimization suggestion. 
unlike the generic SharedReentrancyGuard, the LoanIdGuad is targeted on the loans only. while having it in the common base State contract is convenient, it adds unnecessary code to all the modules.  
pls consider moving it to the LoanMaintenance contract - the nearest contract shared by the LoanOpenings and LoanClosingsWith.